File: 2367. Number of Arithmetic Triplets/solution.js
/**
 * @param {number[]} nums
 * @param {number} diff
 * @return {number}
 */
 var arithmeticTriplets = function(nums, diff) {
  let mySet = new Set(nums);
  let count = 0;
  for (let num of mySet){
    if (mySet.has(num + diff) && mySet.has(num + diff + diff)){
      count += 1;
    }
  }

  return count;
};

File: 2367. Number of Arithmetic Triplets/solution.java
class Solution {
  public int arithmeticTriplets(int[] nums, int diff) {
    Set<Integer> mySet = new HashSet<>();
    for (int index = 0; index < nums.length; index++){
      mySet.add(nums[index]);
    }

    int count = 0;
    for (Integer num : mySet){
      if (mySet.contains(num + diff) && mySet.contains(num + diff + diff)){
        count += 1;
      }
    }

    return count;
  }
}